WordPress/WordPressTest/MediaPicker/MediaLibraryPickerDataSourceTests.swift

```diff
@@ -73,6 +74,44 @@ class MediaLibraryPickerDataSourceTests: CoreDataTestCase {
         self.waitForExpectations(timeout: 5, handler: nil)
     }
 
+    func testMediaGroupUpdates() {
+        contextManager.useAsSharedInstance(untilTestFinished: self)
+        dataSource.setMediaTypeFilter(.image)
+
+        // This variable tracks how many times the album cover (which is what
+        // the "group" is in this use case) has changed.
+        var changes = 0
+        dataSource.registerGroupChangeObserverBlock {
+            changes += 1
+        }
+
+        // Adding a video does not change the album cover.
+        let video = MediaBuilder(context).build()
+        video.remoteStatus = .sync
+        video.blog = self.blog
+        video.mediaType = .video
+        contextManager.saveContextAndWait(context)
+        expect(changes).toEventually(equal(0))
+
+        // Adding a newly created image changes the album cover.
+        let newImage = MediaBuilder(context).build()
+        newImage.remoteStatus = .sync
+        newImage.blog = self.blog
+        newImage.mediaType = .image
+        newImage.creationDate = Date()
+        contextManager.saveContextAndWait(context)
+        expect(changes).toEventually(equal(1))
+
+        // Added an old image does not change the album cover.
+        let oldImage = MediaBuilder(context).build()
+        oldImage.remoteStatus = .sync
+        oldImage.blog = self.blog
+        oldImage.mediaType = .image
+        oldImage.creationDate = Date().advanced(by: -60)
+        contextManager.saveContextAndWait(context)
+        expect(changes).toEventually(equal(1))
```

Comment:
Same observation about the possibility of a false positive. In this case, I think this should work:

```suggestion
        expect(changes).toNever(beGreaterThan(1))
```